operator/main.go

```diff
@@ -148,6 +161,74 @@ func runOperator(cmd *cobra.Command) {
 	}
 	close(k8sInitDone)
 
+	k8sversion.Update(k8s.Client(), option.Config)
+	if !k8sversion.Capabilities().MinimalVersionMet {
+		log.Fatalf("Minimal kubernetes version not met: %s < %s",
+			k8sversion.Version(), k8sversion.MinimalVersionConstraint)
+	}
+
+	// Get hostname for identity name of the lease lock holder.
+	// We identify the leader of the operator cluster using hostname.
+	operatorID, err := os.Hostname()
+	if err != nil {
+		log.WithError(err).Fatal("failed to get hostname")
+	}
+	operatorID = rand.RandomStringWithPrefix(operatorID+"-", 10)
+
+	ns := option.Config.K8sNamespace
+	// If due to any reason the NAMESPACE is not set we assume it to be
+	// in default namespace.
```

Comment:
Nit: previously, it looks like you were using the "NAMESPACE" env var, and this comment is a hold-over?

Reply:
Yeah, I think I should update it to `CILIUM_K8S_NAMESPACE` as this is the environment variable which populates the option config for K8sNamespace.